tket/tests/Circuit/test_Circ.cpp

```diff
@@ -2775,6 +2775,89 @@ SCENARIO("Named operation groups") {
+    Circuit c(10, 1);
+    c.add_op<unsigned>(OpType::H, {0});
+    c.add_op<unsigned>(OpType::T, {1});
+    c.add_op<unsigned>(OpType::S, {2});
+    c.add_op<unsigned>(OpType::X, {3});
+    c.add_op<unsigned>(OpType::Y, {4});
+    c.add_op<unsigned>(OpType::Z, {5});
+    c.add_op<unsigned>(OpType::S, {6});
+    c.add_op<unsigned>(OpType::Z, {7});
+    c.add_op<unsigned>(OpType::V, {8});
+    c.add_op<unsigned>(OpType::H, {9});
+
+    c.add_op<unsigned>(OpType::CX, {0, 1});
+    c.add_op<unsigned>(OpType::CZ, {1, 2});
+    c.add_op<unsigned>(OpType::CY, {2, 3});
+    c.add_op<unsigned>(OpType::CX, {3, 4});
+    c.add_op<unsigned>(OpType::CZ, {4, 5});
+    c.add_op<unsigned>(OpType::CY, {5, 6});
+    c.add_op<unsigned>(OpType::ZZMax, {6, 7});
+    c.add_op<unsigned>(OpType::CX, {7, 8});
+    c.add_op<unsigned>(OpType::CZ, {8, 9});
+
+    c.add_op<unsigned>(OpType::CCX, {0, 1, 2});
+    c.add_op<unsigned>(OpType::CCX, {1, 2, 3});
+    c.add_op<unsigned>(OpType::CCX, {2, 3, 4});
+    c.add_op<unsigned>(OpType::CCX, {3, 4, 5});
+    c.add_op<unsigned>(OpType::CCX, {4, 5, 6});
+    c.add_op<unsigned>(OpType::CCX, {5, 6, 7});
+    c.add_op<unsigned>(OpType::CCX, {6, 7, 8});
+    c.add_op<unsigned>(OpType::CCX, {7, 8, 9});
+
+    c.add_op<unsigned>(OpType::CnX, {0, 1, 2, 3});
+    c.add_op<unsigned>(OpType::CnX, {1, 2, 3, 4});
+    c.add_op<unsigned>(OpType::CnX, {2, 3, 4, 5});
+    c.add_op<unsigned>(OpType::CnX, {3, 4, 5, 6});
+    c.add_op<unsigned>(OpType::CnX, {4, 5, 6, 7});
+    c.add_op<unsigned>(OpType::CnX, {5, 6, 7, 8});
+    c.add_op<unsigned>(OpType::CnX, {6, 7, 8, 9});
+
+    c.add_op<unsigned>(OpType::CnX, {0, 1, 2, 3, 4});
+    c.add_op<unsigned>(OpType::CnX, {1, 2, 3, 4, 5});
+    c.add_op<unsigned>(OpType::CnX, {2, 3, 4, 5, 6});
+    c.add_op<unsigned>(OpType::CnX, {3, 4, 5, 6, 7});
+    c.add_op<unsigned>(OpType::CnX, {4, 5, 6, 7, 8});
+    c.add_op<unsigned>(OpType::CnX, {5, 6, 7, 8, 9});
+
+    c.add_op<unsigned>(OpType::CnX, {0, 1, 2, 3, 4, 5});
+    c.add_op<unsigned>(OpType::CnX, {1, 2, 3, 4, 5, 6});
+    c.add_op<unsigned>(OpType::CnX, {2, 3, 4, 5, 6, 7});
+    c.add_op<unsigned>(OpType::CnX, {3, 4, 5, 6, 7, 8});
+    c.add_op<unsigned>(OpType::CnX, {4, 5, 6, 7, 8, 9});
+
+    c.add_op<unsigned>(OpType::CnX, {0, 1, 2, 3, 4, 5, 6});
+    c.add_op<unsigned>(OpType::CnX, {1, 2, 3, 4, 5, 6, 7});
+    c.add_op<unsigned>(OpType::CnX, {2, 3, 4, 5, 6, 7, 8});
+    c.add_op<unsigned>(OpType::CnX, {3, 4, 5, 6, 7, 8, 9});
+
+    c.add_op<unsigned>(OpType::CnX, {0, 1, 2, 3, 4, 5, 6, 7});
+    c.add_op<unsigned>(OpType::CnX, {1, 2, 3, 4, 5, 6, 7, 8});
+    c.add_op<unsigned>(OpType::CnX, {2, 3, 4, 5, 6, 7, 8, 9});
+
+    c.add_op<unsigned>(OpType::CnX, {0, 1, 2, 3, 4, 5, 6, 7, 8});
+    c.add_op<unsigned>(OpType::CnX, {1, 2, 3, 4, 5, 6, 7, 8, 9});
+
+    c.add_op<unsigned>(OpType::CnX, {0, 1, 2, 3, 4, 5, 6, 7, 8, 9});
+    c.add_barrier({0, 1, 2, 3, 4, 5, 6, 7, 8, 9});
+    c.add_measure(Qubit(0), Bit(0));
```

Comment:
Is measure a one or two qubit gat within this definition?

Reply:
It would be considered a single qubit quantum operation, as it would only have on quantum in edge. However `Measure` gates are also not counted so it should not contribute to the final count.